core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -585,7 +621,12 @@ private void initializeWriter(StoreConfiguration cfg, CacheWriter writer, Initia
       if (writer != null) {
          if (writer instanceof DelegatingCacheWriter)
```

Comment:
Interesting I didn't realize you couldn't have them together.  To me that makes no sense.  That means singleton stores create a single point of failure, which imo is terrible.  @danberindei does that sound correct to you?  What is the use case for singleton store then?